Sources/FileSystem/FileSystem.swift

```diff
@@ -432,10 +480,24 @@ public struct FileSystem: FileSysteming, Sendable {
     }
 
     public func writeText(_ text: String, at path: AbsolutePath, encoding: String.Encoding) async throws {
+        try await writeText(text, at: path, encoding: encoding, options: Set())
+    }
+
+    public func writeText(
+        _ text: String,
+        at path: AbsolutePath,
+        encoding: String.Encoding,
+        options: Set<WriteTextOptions>
+    ) async throws {
         logger?.debug("Writing text at path \(path.pathString).")
         guard let data = text.data(using: encoding) else {
             throw FileSystemError.cantEncodeText(text, encoding)
         }
+
+        if try await exists(path), options.contains(.overriding) {
+            try await remove(path)
+        }
```

Comment:
```suggestion
        if options.contains(.overriding), try await exists(path) {
            try await remove(path)
        }
```

I don't think this syntax specifically is valid but you can make it valid by using a ternary operator, I think. Either way, I'd also run `exists` if we actually need to check for the path. We don't want run this async check even if `overriding` was not specified.